colcon-build-and-test/action.yaml

```
@@ -66,12 +66,15 @@ runs:
     - name: Test
       run: |
         . /opt/ros/${{ inputs.rosdistro }}/setup.sh
-        colcon lcov-result --initial
+        colcon lcov-result --initial \
+          --packages-above ${{ inputs.target-packages }}
         colcon test --event-handlers console_cohesion+ \
           --packages-above ${{ inputs.target-packages }} \
           --return-code-on-test-failure
-        colcon lcov-result
-        colcon coveragepy-result
+        colcon lcov-result \
+          --packages-above ${{ inputs.target-packages }}
+        colcon coveragepy-result \
+          --packages-above ${{ inputs.target-packages }}
```

If you don't plan to add other options, I think this is enough.

```suggestion
        colcon lcov-result --packages-above ${{ inputs.target-packages }}
        colcon coveragepy-result --packages-above ${{ inputs.target-packages }}
```

---

Changed 4d390b1